app/models/concerns/hyku_addons/bulkrax/object_factory_behavior.rb

```diff
@@ -0,0 +1,112 @@
+        object.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX if object.respond_to? :reindex_extent
+        run_callbacks :save do
+          run_callbacks :create do
+            if klass == ::AdminSet
+              create_admin_set(attrs)
+            elsif klass == ::Collection
+              create_collection(attrs)
+            else
+              work_actor.create(environment(attrs))
+            end
+          end
+        end
+        log_created(object)
+      end
+      # rubocop:enable Metrics/MethodLength
+
+      # rubocop:disable Metrics/PerceivedComplexity
+      # rubocop:disable Metrics/CyclomaticComplexity
+      def update
+        raise "Object doesn't exist" unless object
+
+        destroy_existing_files if @replace_files && (klass != ::Collection || klass != ::AdminSet)
+        attrs = update_attributes
+        run_callbacks :save do
+          if klass == ::AdminSet
+            update_admin_set(attrs)
+          elsif klass == ::Collection
+            update_collection(attrs)
+          else
+            work_actor.update(environment(attrs))
+          end
+        end
+        log_updated(object)
+      end
+      # rubocop:enable Metrics/PerceivedComplexity
+      # rubocop:enable Metrics/CyclomaticComplexity
+
+      def create_admin_set(attrs)
+        attrs.delete("collection_type_gid")
+        object.members = members
+        object.attributes = attrs
+        object.save!
+      end
+
+      def update_admin_set(attrs)
+        attrs.delete("collection_type_gid")
+        object.members = members
+        object.attributes = attrs
+        object.save!
+      end
+
+      def find
+        if attributes[:id]
+          find_by_id
+        elsif attributes[system_identifier_field].present? && klass.new.respond_to?(system_identifier_field)
+          search_by_identifier
+        elsif klass == AdminSet && attributes[:title].present?
+          search_by_title_or_identifier
+        end
+      end
+
+      def search_by_title_or_identifier
+        AdminSet.where(title: Array(attributes[:title]).first).first || AdminSet.where(id: Array(attributes[:title]).first).first
+      end
+
+      def permitted_attributes
+        klass.properties.keys.map(&:to_sym) + %i[id edit_users edit_groups read_groups visibility work_members_attributes admin_set_id]
+      end
+
+      # Override if we need to map the attributes from the parser in
+      # a way that is compatible with how the factory needs them.
+      def transform_attributes
+        if klass == ::Collection || klass == ::AdminSet
+          attributes.slice(*permitted_attributes)
+        else
+          attrs = attributes.slice(*permitted_attributes).merge(file_attributes)
+          attrs = attrs.merge("file_set" => attributes["file_set"])
+          attrs["uploaded_files"].each_with_index { |id, i| attrs["file_set"][i]["uploaded_file_id"] = id if attrs["file_set"][i].present? } if attrs["file_set"].present?
```

Comment:
We talked about this line on Slack, and I proposed a refactor.  I think something along the lines of this almost gets there, but it might need a bit more work:
```
file_count = attrs['uploaded_files'].size
attrs["file_set"] = Hash[(0..file_count).zip attrs['uploaded_files']]
````

Reply:
As I dont understand the code fully, or if there are good enough tests around this section to properly refactor it, i went with the simple solution. 

```          attrs["uploaded_files"].each_with_index do |id, i|
            attrs["file_set"][i]["uploaded_file_id"] = id if attrs["file_set"][i].present?
          end if attrs["file_set"].present?
```